streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TimeWindowedKStreamTest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2018 Joan Goyeau.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.scala
+
+import org.apache.kafka.streams.kstream.{Materialized, SessionWindows, TimeWindows}
+import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.Serdes._
+import org.apache.kafka.streams.scala.utils.TestDriver
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FlatSpec, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class TimeWindowedKStreamTest extends FlatSpec with Matchers with TestDriver {
+
+  "Counting a TimeWindowedKStream" should "count correctly records" in {
+    val builder = new StreamsBuilder()
+    val sourceTopic = "source"
+    val sinkTopic = "sink"
+
+    val sessionWindowedStream =
+      builder.stream[String, String](sourceTopic).groupByKey.windowedBy(TimeWindows.of(1000))
+    sessionWindowedStream.count().toStream((windowed, _) => windowed.key).to(sinkTopic)
+
+    val testDriver = createTestDriver(builder)
+
+    testDriver.pipeRecord(sourceTopic, ("1", "value1"))
+    testDriver.readRecord[String, Long](sinkTopic).value shouldBe 1
+
+    testDriver.advanceWallClockTime(500)
+    testDriver.pipeRecord(sourceTopic, ("1", "value2"))
+    testDriver.readRecord[String, Long](sinkTopic).value shouldBe 2
+
+    testDriver.advanceWallClockTime(700)
+    testDriver.pipeRecord(sourceTopic, ("1", "value3"))
+    testDriver.readRecord[String, Long](sinkTopic).value shouldBe 2
```

Comment:
Do you guys understand why this is 3 instead of 2?
The window should ditch the first event no?

Reply:
It turns out the `advanceWallClockTime` is not working